lib/include/elements/element/gallery/button.hpp

```diff
@@ -231,26 +240,97 @@ namespace cycfi { namespace elements
       float                   _radius;
    };
 
+   template<typename Base, default_button_styler::direction Direction>
+   struct button_styler_rounded_half : Base
+   {
+      using base_type = button_styler_rounded_half<typename Base::base_type, Direction>;
+      using direction_t = default_button_styler::direction;
+
+                              button_styler_rounded_half(Base base, float r)
+                               : Base(std::move(base)), _radius(r)
+                              {}
+
+      virtual float           get_corner_radius_top_left() const override;
+      virtual float           get_corner_radius_top_right() const override;
+      virtual float           get_corner_radius_bottom_left() const override;
+      virtual float           get_corner_radius_bottom_right() const override;
+
+   private:
+
+      float                   _radius;
+   };
+
+   template<typename Base, default_button_styler::corner Corner>
+   struct button_styler_rounded_corner : Base
+   {
+      using base_type = button_styler_rounded_corner<typename Base::base_type, Corner>;
+      using corner_t = default_button_styler::corner;
+
+                              button_styler_rounded_corner(Base base, float r)
+                               : Base(std::move(base)), _radius(r)
+                              {}
+
+      virtual float           get_corner_radius_top_left() const override;
+      virtual float           get_corner_radius_top_right() const override;
+      virtual float           get_corner_radius_bottom_left() const override;
+      virtual float           get_corner_radius_bottom_right() const override;
+
+   private:
+
+      float                  _radius;
+   };
+
+   template <typename Base>
+   struct button_styler_with_individual_corner_radii : Base
+   {
+      using base_type = button_styler_with_individual_corner_radii<typename Base::base_type>;
+
+                              button_styler_with_individual_corner_radii(Base base, float top_left, float top_right, float bottom_right, float bottom_left)
```

Comment:
Please limit to 80 characters :-)

Reply:
While I didn't quite manage to keep it below 80 characters, I've now reduced it to be consistent with the size of the existing code (longest is ~100 chars).

Reply:
> While I didn't quite manage to keep it below 80 characters, I've now reduced it to be consistent with the size of the existing code (longest is ~100 chars).

Yes, it's not a strict rule, but more like a guideline. ~100 chars is fine.